templates/OpenEdx/0/docker-compose.yml

```diff
@@ -0,0 +1,138 @@
+     - /dev/log:/dev/log
+    volume_driver: ${volume_driver}
+    command: |
+        bash -c "rm -rf /edx/app/edx_ansible/edx_ansible; git clone https://github.com/OpenSaasAU/edx-configuration /edx/app/edx_ansible/edx_ansible; cd /edx/app/edx_ansible/edx_ansible/docker/plays; rm -f /edx/app/supervisor/conf.d/*; /edx/app/edx_ansible/venvs/edx_ansible/bin/ansible-playbook edxapp.yml -i '127.0.0.1,' -c local -e '{\"service_variants_enabled\":[\"cms\"]}' -e \" EDXAPP_PLATFORM_NAME='${EDXAPP_PLATFORM_NAME}' EDXAPP_SITE_NAME='${EDX_DOMAIN_NAME}' EDXAPP_CMS_SITE_NAME='${STUDIO_DOMAIN_NAME}' EDXAPP_LMS_NGINX_PORT='${EXPOSED_PORT}' EDXAPP_LMS_PREVIEW_NGINX_PORT='${EXPOSED_PORT}' EDXAPP_CMS_NGINX_PORT='${EXPOSED_PORT}' EDXAPP_CELERY_USER='${CELERY_EDXAPP_USER}' EDXAPP_CELERY_PASSWORD='${CELERY_EDXAPP_PASSWORD}' EDXAPP_MYSQL_USER='${MYSQL_EDXAPP_USER}' EDXAPP_MYSQL_PASSWORD='${MYSQL_EDXAPP_PASSWORD}' EDXAPP_MYSQL_PASSWORD_ADMIN='${MYSQL_ROOT_PASSWORD}' EDXAPP_MONGO_USER='${MONGO_EDXAPP_USER}' EDXAPP_MONGO_PASSWORD='${MONGO_EDXAPP_PASSWORD}'\" -t 'install:app-requirements,install:base,install:configuration' -e@/edx/app/edx_ansible/edx_ansible/docker/build/edxapp/rancher_overrides.yml; /edx/app/supervisor/venvs/supervisor/bin/supervisord -n --configuration /edx/app/supervisor/supervisord.conf"
+
+edxworker:
+    labels:
+        io.rancher.container.pull_image: always
+    image: opensaas/edxapp:trusty-v3
+    volumes:
+     - ${EDXAPP_PLATFORM_ROOT}:/edx/app/edxapp/edx-platform
+     - /dev/log:/dev/log
+    volume_driver: ${volume_driver}
+    command: |
+        bash -c "rm -rf /edx/app/edx_ansible/edx_ansible; git clone https://github.com/OpenSaasAU/edx-configuration /edx/app/edx_ansible/edx_ansible; cd /edx/app/edx_ansible/edx_ansible/docker/plays; rm -f /edx/app/supervisor/conf.d/*; /edx/app/edx_ansible/venvs/edx_ansible/bin/ansible-playbook edxapp.yml -i '127.0.0.1,' -c local -e '{\"service_variants_enabled\":[\"cms\",\"lms\"]}' -e \"EDXAPP_PLATFORM_NAME='${EDXAPP_PLATFORM_NAME}' EDXAPP_SITE_NAME='${EDX_DOMAIN_NAME}' EDXAPP_CMS_SITE_NAME='${STUDIO_DOMAIN_NAME}' EDXAPP_LMS_NGINX_PORT='${EXPOSED_PORT}' EDXAPP_LMS_PREVIEW_NGINX_PORT='${EXPOSED_PORT}' EDXAPP_CMS_NGINX_PORT='${EXPOSED_PORT}' EDXAPP_CELERY_USER='${CELERY_EDXAPP_USER}' EDXAPP_CELERY_PASSWORD='${CELERY_EDXAPP_PASSWORD}' EDXAPP_MYSQL_USER='${MYSQL_EDXAPP_USER}' EDXAPP_MYSQL_PASSWORD='${MYSQL_EDXAPP_PASSWORD}' EDXAPP_MYSQL_PASSWORD_ADMIN='${MYSQL_ROOT_PASSWORD}' EDXAPP_MONGO_USER='${MONGO_EDXAPP_USER}' EDXAPP_MONGO_PASSWORD='${MONGO_EDXAPP_PASSWORD}' celery_worker='true'\" -t 'install:app-requirements,install:base,install:configuration' -e@/edx/app/edx_ansible/edx_ansible/docker/build/edxapp/rancher_overrides.yml; /edx/app/supervisor/venvs/supervisor/bin/supervisord -n --configuration /edx/app/supervisor/supervisord.conf"
+        
+edxappMigrate:
+    image: opensaas/edxapp:trusty-v3
+    volumes:
+     - ${EDXAPP_PLATFORM_ROOT}:/edx/app/edxapp/edx-platform

```

Comment:
@borisno2 Since lms, cms, and edxworker are sharing the same storage space EDXAPP_PLATFORM_ROOT, it will be more efficient if you use sidekicks to manage to start these services on the same host, because the current implementation enforces user to use shared hosting since there is no guarantee that the services will start on the same host.